cmd/soroban-cli/src/commands/tx/edit.rs

```diff
@@ -0,0 +1,186 @@
+use std::{
+    env,
+    fs::{self},
+    io::{stdin, Cursor, IsTerminal, Write},
+    path::PathBuf,
+    process::{self},
+};
+
+use serde_json::json;
+use stellar_xdr::curr;
+
+use crate::{commands::global, print::Print};
+
+const DEFAULT_JSON: &str = r#"{
+  "tx": {
+    "tx": {
+      "source_account": "",
+      "fee": 100,
+      "seq_num": 0,
+      "cond": "none",
+      "memo": "none",
+      "operations": [],
+      "ext": "v0"
+    },
+    "signatures": []
+  }
+}
+"#;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+
+    #[error(transparent)]
+    StellarXdr(#[from] stellar_xdr::curr::Error),
+
+    #[error(transparent)]
+    SerdeJson(#[from] serde_json::Error),
+
+    #[error(transparent)]
+    Base64Decode(#[from] base64::DecodeError),
+
+    #[error("Editor returned non-zero status")]
+    EditorNonZeroStatus,
+}
+
+// Command to edit the transaction
+/// e.g. `stellar tx new manage-data --data-name hello --build-only | stellar tx edit`
+#[derive(Debug, clap::Parser, Clone, Default)]
+#[group(skip)]
+pub struct Cmd {}
+
+impl Cmd {
+    pub fn run(&self, global_args: &global::Args) -> Result<(), Error> {
+        let print = Print::new(global_args.quiet);
+        let json: String = if stdin().is_terminal() {
+            DEFAULT_JSON.to_string()
+        } else {
+            let mut input = String::new();
+            stdin().read_line(&mut input)?;
+            let input = input.trim();
+            xdr_to_json::<curr::TransactionEnvelope>(input)?
+        };
+
+        let path = tmp_file(&json)?;
+        let editor = get_editor();
+
+        open_editor(&print, &editor, &path)?;
+
+        let contents = fs::read_to_string(&path)?;
+        let xdr = json_to_xdr::<curr::TransactionEnvelope>(&contents)?;
+        fs::remove_file(&path)?;
+
+        println!("{xdr}");
+
+        Ok(())
+    }
+}
+
+struct Editor {
+    cmd: String,
+    source: String,
+    args: Vec<String>,
+}
+
+fn tmp_file(contents: &str) -> Result<PathBuf, Error> {
+    let temp_dir = env::current_dir().unwrap_or(env::temp_dir());
```

Comment:
This leaves tmp files in the working directory.
IMO, we should only use temp_dir

Reply:
That's the idea. If the command crashes for some reason, or if you kill the process, you can still have access to the file.